src/compile/data/filter.ts

```diff
@@ -22,4 +22,9 @@ export class FilterNode extends DataFlowNode {
       expr: this.expr
     };
   }
+
+  public hash() {
+    // FIX include something from model in hash
```

Comment:
Hmm I thought we agreed to decouple SelectionFilter from ExpressionFilter. 
So we only merge ExpressionFilter, not SelectionFilter.  

That said, the current code means that it will always be merged.  The question is whether this is the right behavior. Maybe it's fine. 

To test it, create an spec where the same selection is applied two layers and see if merging the results still produce correct output.  (Maybe concat one unit spec with a layered spec then apply selection from the unit spec to both layers in the other spec.)

Reply:
Actually, modify interactive_stocks_nearest_index to use timeUnit would demonstrate if this works well.

Reply:
The sad news is that adding `timeUnit` to `interactive_stocks_nearest_index` doesn't even work, but the diff that this PR is causing to that version seems reasonable.

Thus, I wonder if we should ignore this case for now.  

@domoritz  any thoughts?


In any case, @invokesus  we better remove `// FIX include something from model in hash`.  I don't think we should add stuff from the model to the hash, but rather get rid of model from Filter https://github.com/vega/vega-lite/issues/4084